pandas/core/indexes/base.py

```diff
@@ -264,6 +264,9 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
                 return cls._simple_new(data, name)
 
         from .range import RangeIndex
+        if isinstance(data, ABCPandasArray):
```

Comment:
shouldn't this be in the ExtensionArray section? (like you do in the internals unwrapping)? maybe make this a function that you call from both?

Reply:
isn't this just extract_arrays?

Reply:
ok, still I think you could easily add this to that function, to avoid litering checks like this around.

Reply:
So it turns out we can just call `extract_array` here with `extract_numpy=True`... That wasn't my original intent for the method, but I think it'll be OK.

Reply:
Ahhh never mind, we can't call it since that would *also* unwrap the array from a Series / Index, which we don't want to do at this point.

Reply:
shouldn't this then be ``.to_numpy()`` this is peering into an internal property